File: BinaryTree/js/breadthFirstValues.js
class Node{
    constructor(val){
        this.val = val
        this.left = null
        this.right = null
    }
}

const breadThFirstValues =(root)=>{
    if(root === null) return []

    const queue = [root]
    const values = []

    while(queue.length > 0 ){
        const node = queue.shift()
        values.push(node.val)

        if(node.left) queue.push(node.left)
        if(node.right) queue.push(node.right)
    }

    return values
}

File: BinaryTree/python/breadthFirstValues.py
from collections import deque
from typing import List, Optional

from node import Node


def breadth_first_values(root: Optional[Node]) -> List[int]:
    if root is None:
        return []

    queue = deque([root])
    result: List[int] = []

    while queue:
        current = queue.popleft()
        result.append(current.val)
        if current.left:
            queue.append(current.left)
        if current.right:
            queue.append(current.right)

    return result
